NuRadioReco/eventbrowser/index.py

```diff
@@ -170,46 +180,63 @@ def get_page_content(selection):
     return []
 
 
-# next/previous buttons
+# event selector - couples slider, next/previous button and dropdown input
 @app.callback(
-    Output('event-counter-slider', 'value'),
+    [Output('event-counter-slider', 'value'),
+     Output('event-info-run', 'value'),
+     Output('event-info-id', 'value')],
     [Input('btn-next-event', 'n_clicks_timestamp'),
      Input('btn-previous-event', 'n_clicks_timestamp'),
+     Input('event-info-id', 'value'),
      Input('event-click-coordinator', 'children'),
-     Input('filename', 'value')],
-    [State('event-counter-slider', 'value'),
+     Input('filename', 'value'),
+     Input('event-counter-slider', 'value')],
+    [State('event-info-run', 'value'),
      State('user_id', 'children')]
 )
-def set_event_number(next_evt_click_timestamp, prev_evt_click_timestamp, j_plot_click_info, filename, i_event,
-                     juser_id):
+def set_event_number(next_evt_click_timestamp, prev_evt_click_timestamp, event_id, j_plot_click_info, filename, 
+                     i_event, run_number, juser_id):
     context = dash.callback_context
     if filename is None:
-        return 0
+        return 0, None, None
+    user_id = json.loads(juser_id)
+    number_of_events = provider.get_arianna_io(user_id, filename).get_n_events()
+    event_ids = provider.get_arianna_io(user_id, filename).get_event_ids()
     if context.triggered[0]['prop_id'] == 'filename.value':
-        return 0
+        return 0, event_ids[0][0], event_ids[0][1]
     if context.triggered[0]['prop_id'] == 'event-click-coordinator.children':
         if context.triggered[0]['value'] is None:
-            return 0
-        return json.loads(context.triggered[0]['value'])['event_i']
+            return 0, event_ids[0][0], event_ids[0][1]
+        event_i = json.loads(context.triggered[0]['value'])['event_i']
+        return event_i, event_ids[event_i][0], event_ids[event_i][1]
+    if context.triggered[0]['prop_id'] == 'event-counter-slider.value':
+        return i_event, event_ids[i_event][0], event_ids[i_event][1]
     else:
+        if context.triggered[0]['prop_id'] == 'event-info-id.value':
+            mask = event_ids == (run_number, event_id)
+            event_i = np.where(mask[:,0] * mask[:,1])[0][0]
+            return event_i, run_number, event_id
         if context.triggered[0]['prop_id'] != 'btn-next-event.n_clicks_timestamp' and context.triggered[0]['prop_id'] != 'btn-previous-event.n_clicks_timestamp':
-            return 0
+            return 0, event_ids[0][0], event_ids[0][1]
         if context.triggered[0]['prop_id'] == 'btn-previous-event.n_clicks_timestamp':
             if i_event == 0:
-                return 0
+                event_i = 0
             else:
-                return i_event - 1
+                event_i = i_event - 1
         if context.triggered[0]['prop_id'] == 'btn-next-event.n_clicks_timestamp':
-            user_id = json.loads(juser_id)
+            # user_id = json.loads(juser_id)
```

Comment:
Here (and further below) there are a bunch on uncommented lines. Are they here for a reason (work in progress) or should they rather go?